airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -293,6 +318,61 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     }
   }
 
+  private SyncCheckConnectionFailure checkConnections(GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final JobRunConfig jobRunConfig = jobInputs.getJobRunConfig();
+    final StandardSyncInput syncInput = jobInputs.getSyncInput();
+    final JsonNode sourceConfig = syncInput.getSourceConfiguration();
+    final JsonNode destinationConfig = syncInput.getDestinationConfiguration();
+    final IntegrationLauncherConfig sourceLauncherConfig = jobInputs.getSourceLauncherConfig();
+    final IntegrationLauncherConfig destinationLauncherConfig = jobInputs.getDestinationLauncherConfig();
+    final Long jobId = jobRunConfig.getJobId() != null ? Long.valueOf(jobRunConfig.getJobId()) : 1L;
+    final Integer attemptId = jobRunConfig.getAttemptId() != null ? Math.toIntExact(jobRunConfig.getAttemptId()) : 1;
+    SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobId, attemptId);
+
+    final int attemptCreationVersion =
+        Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
+
+    if (attemptCreationVersion >= CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+      final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(sourceConfig);
+      final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+      final String launchSourceDockerImage = sourceLauncherConfig.getDockerImage();
+
+      if (WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB.equals(launchSourceDockerImage) || checkFailure.isFailed()) {
+        log.info("SOURCE CHECK: Skipped");
+      } else {
+        log.info("SOURCE CHECK: Starting");
+        final StandardCheckConnectionOutput sourceCheckResponse = runMandatoryActivityWithOutput(checkActivity::run, checkSourceInput);
+        if (sourceCheckResponse != null && sourceCheckResponse.getStatus() == Status.FAILED) {
```

Comment:
This logic means that if `sourceCheckResponse == null`, then we will mark the source check as successful. Is this desired? Do we ever expect this to be null? If not then maybe this should throw an exception rather than just silently succeeding?

Reply:
I think that in the user-facing happy-path, it's a safe bet that the connection was previously checked at least once (in the UI, when setting up the source/destination), so I don't think a `null` will occur.  However, in some tests it was null so I chose to count those as a "passing" check to the sync could continue.

Reply:
I'm not sure I understand what you mean by this:
>  it's a safe bet that the connection was previously checked at least once (in the UI, when setting up the source/destination), so I don't think a null will occur
This `sourceCheckResponse` object is initialized in the line just before this, when we run the check activity, so I don't think checking the connection when setting up the source/destination has any effect on that. I was more asking if there is any reason we should expect that variable to be initialized to `null`.

Reply:
Ah! Sorry I misinterpreted that.  I can no longer remember what terrible thing I used to be doing where `null` could be returned from those activities... perhaps it was suppressing a more severe error with the activity failing in a test.  It doesn't seem to be happening now, so I've removed the null checks via 9a348ad49f